server/src/main/java/org/opensearch/cluster/routing/allocation/RebalanceConstraints.java

```diff
@@ -27,9 +29,13 @@ public class RebalanceConstraints {
 
     private Map<String, Constraint> constraints;
 
-    public RebalanceConstraints() {
+    public RebalanceConstraints(float preferPrimaryBalanceBuffer) {
         this.constraints = new HashMap<>();
         this.constraints.putIfAbsent(INDEX_PRIMARY_SHARD_BALANCE_CONSTRAINT_ID, new Constraint(isPerIndexPrimaryShardsPerNodeBreached()));
+        this.constraints.putIfAbsent(
```

Comment:
this will override the same key since the values for both INDEX_PRIMARY_SHARD_BALANCE_CONSTRAINT_ID and CLUSTER_PRIMARY_SHARD_REBALANCE_CONSTRAINT_ID are same.